core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPostProcessing.kt

```diff
@@ -120,106 +122,142 @@ class STDCMPostProcessing(private val graph: STDCMGraph) {
+        val plannedNodes = getPlannedNodes(nodes)
+        var delayForPlannedNodes = 0.0
+        for (plannedNode in plannedNodes) {
+            val timeDiff = plannedNode.getTimeDiffWithPlannedArrival(addedDelay)
+            // Train passes before planned node: adding delay can make the train pass closer to
+            // planned arrival time
+            if (timeDiff < 0.0) {
+                delayForPlannedNodes = abs(timeDiff)
+                break
+            }
+        }
+        // Total delay added shouldn't be higher than the maximum delay that can possibly be added
+        // to the nodes.
+        return min(addedDelay + delayForPlannedNodes, maxNodeDelay)
+    }
+
+    private fun getNodes(edges: List<STDCMEdge>): List<STDCMNode> {
+        val nodes = edges.map { it.previousNode }.toMutableList()
+        nodes.add(edges.last().getEdgeEnd(graph))
+        return nodes
+    }
+
+    private fun getPlannedNodes(nodes: List<STDCMNode>): List<STDCMNode> {
+        return nodes.filter { it.plannedTimingData != null }
+    }
+
+    private fun getMaxNodeDelay(nodes: List<STDCMNode>): Double {
+        return nodes.minOfOrNull { it.maximumAddedDelay + it.totalPrevAddedDelay }!!
+    }
+
+    /** Builds the list of stops from the edges */
+    private fun makeStops(edges: List<STDCMEdge>): List<TrainStop> {
+        val res = ArrayList<TrainStop>()
+        var offset = 0.meters
+        for (edge in edges) {
+            val prevNode = edge.previousNode
+            // Ignore first path node and last node (we aren't checking lastEdge.getEdgeEnd())
+            if (
+                prevNode.previousEdge != null &&
+                    prevNode.stopDuration != null &&
+                    prevNode.stopDuration >= 0
             )
-        offset += edge.length.distance
+                res.add(
+                    TrainStop(
+                        offset.meters,
+                        prevNode.stopDuration,
+                        // TODO: forward and use onStopSignal param from request
+                        isTimeStrictlyPositive(prevNode.stopDuration)
+                    )
+                )
+            offset += edge.length.distance
+        }
+        return res
     }
-    return res
-}
 
-/** Builds the list of block ranges, merging the ranges on the same block */
-private fun makeBlockRanges(edges: List<STDCMEdge>): List<PathfindingEdgeRangeId<Block>> {
-    val res = ArrayList<PathfindingEdgeRangeId<Block>>()
-    var i = 0
-    while (i < edges.size) {
-        val edge = edges[i]
-        val start = edge.envelopeStartOffset
-        var length = edge.length
-        while (i + 1 < edges.size) {
-            val nextEdge = edges[i + 1]
-            if (edge.block != nextEdge.block) break
-            length += nextEdge.length.distance
+    /** Builds the list of block ranges, merging the ranges on the same block */
+    private fun makeBlockRanges(edges: List<STDCMEdge>): List<PathfindingEdgeRangeId<Block>> {
```

Comment:
The indentation change makes it hard to review: I'll assume that nothing change in methods that aren't directly relevant. If there's other changes please let me know

Reply:
Nope just moved 'em into the class, no other changes. Didn't see why we'd need the private methods outside of the class tbh.

Reply:
BTW, there is an option to hide whitespace changes on github. 

![image](https://github.com/user-attachments/assets/ff84ecde-0d02-44f3-9f7d-3efa255e0926)